### PR TITLE
feat: Customize the readiness probe of APISIX pod (#545)

### PR DESCRIPTION
**Pull request by joyyir**
Please refer #545

### CODE REVIEW

File: charts/apisix/values.yaml [hunk 1]
```diff
@@ -157,6 +157,16 @@ apisix:
...
   extraEnvVars: []
 
+  # -- Set the readinessProbe for Apache APISIX pods
+  readinessProbe:
+    failureThreshold: 6
+    initialDelaySeconds: 10
+    periodSeconds: 10
+    successThreshold: 1
+    tcpSocket:
+      port: &gatewayHttpContainerPort 9080
```
Comment (tao12345666333) [inline charts/apisix/values.yaml:167]: This is incorrect.

File: charts/apisix/values.yaml [hunk 2]
```diff
@@ -232,7 +242,7 @@ gateway:
   http:
     enabled: true
     servicePort: 80
-    containerPort: 9080
+    containerPort: *gatewayHttpContainerPort
```
Comment (tao12345666333) [inline charts/apisix/values.yaml:245]: ditto

**Review by tao12345666333 — CHANGES_REQUESTED** [review]

File: charts/apisix/values.yaml (hunk 1 above)
Comment (joyyir) [reply charts/apisix/values.yaml:167]: I used `YAML Anchors` to refer to the same values.
Wouldn't you recommend it?
https://helm.sh/docs/chart_template_guide/yaml_techniques/#yaml-anchors.

Comment (tao12345666333) [reply charts/apisix/values.yaml:167]: Sorry, this does work. I may have made a mistake during my previous verification.

**Review by tao12345666333 — APPROVED** [review]
LGTM

**Review by lingsamuel — APPROVED** [review]

**Review by Gallardot — APPROVED** [review]

**Pull request merged**